lms/djangoapps/class_dashboard/tests/test_dashboard_data.py

```diff
@@ -85,6 +98,36 @@ def setUp(self):
                     state=json.dumps({'attempts': self.attempts}),
                 )
 
+            # adding an instructor and a staff to the site. They should not be counted. Test should not fail
+            StudentModuleFactory.create(
+                grade=1,
+                max_grade=1,
+                student=instructor_member,
+                course_id=self.course.id,
+                module_state_key=self.item.location,
+                state=json.dumps({'attempts': self.attempts}),
+            )
+
+            StudentModuleFactory.create(
+                grade=1,
+                max_grade=1,
+                student=staff_member,
+                course_id=self.course.id,
+                module_state_key=self.item.location,
+                state=json.dumps({'attempts': self.attempts}),
+            )
+
+            # Adding this student will fail the test
+
+            # StudentModuleFactory.create(
+            #     grade=1,
+            #     max_grade=1,
+            #     student=newstudent,
+            #     course_id=self.course.id,
+            #     module_state_key=self.item.location,
+            #     state=json.dumps({'attempts': self.attempts}),
+            # )
+
             for j, user in enumerate(self.users):
                 StudentModuleFactory.create(
```

Comment:
This was not in your code, @lydiali, but this is a bit sloppy.

1st, it's missing the `student=user` argument.  2nd, it's `module_state_key=` argument should not be set to `self.item.location`, as that's not a sequential.  Rather, `self.sub_section.location` is the sequential location.  I hope that doesn't break any existing tests, but it's a bug that should be fixed.